crates/client-api/src/routes/database.rs

```diff
@@ -92,23 +92,37 @@ pub async fn call<S: ControlStateDelegate + NodeDelegate>(
             host.spawn_module_host(dbic).await.map_err(log_and_500)?
         }
     };
-    let result = match module.call_reducer(caller_identity, None, &reducer, args).await {
-        Ok(rcr) => rcr,
-        Err(e) => {
-            let status_code = match e {
-                ReducerCallError::Args(_) => {
-                    log::debug!("Attempt to call reducer with invalid arguments");
-                    StatusCode::BAD_REQUEST
-                }
-                ReducerCallError::NoSuchModule(_) => StatusCode::NOT_FOUND,
-                ReducerCallError::NoSuchReducer => {
-                    log::debug!("Attempt to call non-existent reducer {}", reducer);
-                    StatusCode::NOT_FOUND
+
+    let result = if let Err(e) = module.call_identity_connected_disconnected(caller_identity, true).await {
+        return Err((StatusCode::NOT_FOUND, format!("{:#}", anyhow::anyhow!(e))).into());
+    } else {
```

Comment:
This `if`/`else` chain is kinda grody, and I don't think it's necessary. Can this be:

```rust
if let Err(e) = module.call_identity_connected_disconnected(caller_identity, true).await {
    // handle error
}
let result = match module.call_reducer(whatever).await {
    // whatever
};
```

It also seems like the `if let Err(e) = foo { return Err(whatever); }` could be made into a helper function, something like:

```rust
fn with_status_code(code: StatusCode, err: impl std::string::ToString) -> impl Axum::IntoResponse {
    (code, err.to_string())
}

// handling an error
let result = module.call_reducer(caller_identity, None, &reducer, args)
    .map_err(|e| with_status_code(StatusCode::NOT_FOUND, e)?;
```